### PR TITLE
chore: move filter manager from status-go to go-waku

### PR DESCRIPTION
**Pull request by chaitanyaprem**
- [x] migrate filter-manager from status-go to api
- [x] Write tests for filter manager

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]

File: waku/v2/api/filter/filter_manager.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		mgr.Lock()
+		if len(mgr.waitingToSubQueue) > 0 {
+			for af := range mgr.waitingToSubQueue {
+				// TODO: change the below logic once topic specific health is implemented for lightClients
+				if pubsubTopic == "" || pubsubTopic == af.contentFilter.PubsubTopic {
+					// check if any filter subs are pending and subscribe them
+					mgr.logger.Debug("subscribing from filter queue", zap.String("filter-id", af.ID), zap.Stringer("content-filter", af.contentFilter))
+					go mgr.subscribeAndRunLoop(af)
+				} else {
+					// TODO: Can this cause issues?
```
Comment (kaichaosun) [inline waku/v2/api/filter/filter_manager.go:185]: Is this comment still valid?

File: waku/v2/api/filter/filter_manager.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+					go mgr.subscribeAndRunLoop(af)
+				}
+				mgr.incompleteFilterBatch = make(map[string]filterConfig)
+				mgr.Unlock()
+			}
+		}
+	}
+}
+
+/*
```
Comment (kaichaosun) [inline waku/v2/api/filter/filter_manager.go:104]: Better be consistent with comment style.

**Review by kaichaosun — APPROVED** [review]

File: waku/v2/api/filter/filter_manager.go (hunk 1 above)
Comment (chaitanyaprem) [reply waku/v2/api/filter/filter_manager.go:185]: I don't think so, removing it.

**Comment by chaitanyaprem** [comment]
@richard-ramos any idea why storev3 tests keep failing with some docker-compose not found error.

**Comment by chaitanyaprem** [comment]
Merging as tests passed locally and also did some dogfooding with status-desktop.
Also, sure that CI issue is due to something else entirely.

**Pull request merged**